tests/test_run_plugin.py

```diff
@@ -429,6 +429,17 @@ def test_smashing_f5(tmp_path, wait_until, use_after_idle):
     assert rest == "Hello\nThe process completed successfully."
 
 
+def test_pause_resume_button(tmp_path, wait_until):
+    (tmp_path / "sleeper.py").write_text("import time; print('before'); time.sleep(0.5); print('after')")
+    no_terminal.run_command(f"{utils.quote(sys.executable)} sleeper.py", tmp_path)
+    wait_until(lambda: "before" in get_output())
+    no_terminal.runner.pause_button.event_generate("<Button-1>")
+    assert "after" not in get_output()
+    time.sleep(1)
```

Comment:
- Maybe these two should be in reverse? Even if the pause didn't work, it would be very unlikely that `"after"` appears immediately.
- You need to replace `time.sleep(1)` with something that lets the event loop run, so that if the button doesn't work, text will actually get inserted into the output area. For example:
    ```
    sleep_end = time.time() + 1
    wait_until(lambda: time.time() > sleep_end)
    ```

Reply:
👍

By the way, I assume this is just a mistake in `test_stop_button`:

```python
wait_until(lambda: "started\nKilled.")
```

Reply:
Woah, yes that is a mistake :) This is the problem with writing asserts, they aren't supposed to fail and you can't test them failing easily.